clang-tools-extra/clang-tidy/performance/UnnecessaryValueParamCheck.cpp

```diff
@@ -63,7 +65,11 @@ void UnnecessaryValueParamCheck::registerMatchers(MatchFinder *Finder) {
   Finder->addMatcher(
       traverse(
           TK_AsIs,
-          functionDecl(hasBody(stmt()), isDefinition(), unless(isImplicit()),
+          functionDecl(hasBody(IsAllowedInCoroutines
+                                   ? stmt()
+                                   : static_cast<StmtMatcher>(
+                                         unless(coroutineBodyStmt()))),
```

Comment:
I think we can just write shorter like this: (formatting is wrong here)
```suggestion
          functionDecl(hasBody(IsAllowedInCoroutines ?
                             stmt() : stmt(unless(coroutineBodyStmt()))),
```
no need for static cast and using decl.

Reply:
Note that I wrapped `unless(coroutineBodyStmt())` in a `stmt()` so the resulting matcher would be `stmt(unless(coroutineBodyStmt()))`. Full line would be like this:
```cpp
hasBody(IsAllowedInCoroutines
                 ? stmt()
                 : stmt(unless(coroutineBodyStmt())))
```

Reply:
For formatting par,t my initial suggestion had bad formatting so committing it from github would give CI failure